src/hotspot/share/cds/filemap.cpp

```diff
@@ -1453,20 +1511,25 @@ bool FileMapInfo::open_for_read() {
 void FileMapInfo::open_for_write() {
   LogMessage(cds) msg;
   if (msg.is_info()) {
-    msg.info("Dumping shared data to file: ");
+    if (CDSConfig::is_dumping_preimage_static_archive()) {
+      msg.info("Writing binary AOTConfiguration file: ");
+    } else {
+      msg.info("Dumping shared data to file: ");
+    }
     msg.info("   %s", _full_path);
   }
 
 #ifdef _WINDOWS  // On Windows, need WRITE permission to remove the file.
-    chmod(_full_path, _S_IREAD | _S_IWRITE);
+  chmod(_full_path, _S_IREAD | _S_IWRITE);
 #endif
 
   // Use remove() to delete the existing file because, on Unix, this will
   // allow processes that have it open continued access to the file.
   remove(_full_path);
-  int fd = os::open(_full_path, O_RDWR | O_CREAT | O_TRUNC | O_BINARY, 0444);
+  int mode = CDSConfig::is_dumping_preimage_static_archive() ? 0666 : 0444;
```

Comment:
Why do we need to give different access permission for preimage file compared to other dumping modes?

Reply:
It's to keep the current behavior: -XX:AOTConfiguration creates a file with read/write permission but -XX:AOTCache creates a read-only file.